RedHat-UX/red-hat-design-system
# feat(subnav): add `<rh-subnav>`

## What I did

1. Add `<rh-subnav>`
2. Closes #703 

## Todo
- [x] finalize styles
- [x] implement roving tab index
- [ ] add spec tests

## Testing instructions
- See [deploy preview demo](https://deploy-preview-718--red-hat-design-system.netlify.app/components/subnav/demo/)
- Test in different browser resolutions (ie mobile)

## Reviewer Notes
Ensure correct token usage, and please look for any accessibility issues.
